### ISSUE TITLE
LS + LA Anzahl Pro Tag Counter in Picasso

### ISSUE THREAD

**Issue by feigenmann** [ecamp/ecamp3#625]
Die vorgegebene Anzahl von LS und LA Blocks sollte bei jedem Tag angezeigt werden:
- tagesübergreifende Aktivität
-LA Stunden pro Tag
-LS Stunden pro Tag

**Comment by bb1950328** [comment]
So ein Feature, um die Kontrolle der J+S-Richtlinien zu automatisieren, würde ich sehr begrüssen. Gerade bei der Zählung der J+S-Stunden ist es aber nicht ganz so einfach wie es scheint. Ich habe leider gerade kein Dokument zur Hand, um dies zu belegen, aber mir wurde mal gesagt, dass die J+S-Aktivitäten innerhalb der "normalen" Tageszeiten liegen müssen, damit sie zählen. Ein LS von 01:00-02:00 würde also nicht zählen. Von wann bis wann diese "normalen" Tageszeiten genau sind, ist aber offenbar nicht genau definiert.

**Comment by chrisgwerder** [comment]
J+S hat diese Anforderungen an ein Lagerprogramm was die Zeiten betrifft:
<img width="758" alt="image" src="https://github.com/user-attachments/assets/8bef1a26-4bfa-4a3e-8af1-7779a68e1829">
 
Quelle: https://backend.jugendundsport.ch/fileservice/sdweb-docs-prod-jugdsport-files/files/2024/03/06/8b65a586-b642-44f8-920f-4e665fee56b4.pdf

**Comment by Im-a-Train** [comment]
Als J+S Coach würde ich dieses Feature sehr begrüssen.
Ich denke wir müssen nicht alle Edge-Cases abdecken.

Wenn ich nicht täusche gilt die Regel von 

> J+S hat diese Anforderungen an ein Lagerprogramm was die Zeiten betrifft: <img alt="image" width="758" src="https://private-user-images.githubusercontent.com/319960/369694937-8bef1a26-4bfa-4a3e-8af1-7779a68e1829.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzMyNTg2MzcsIm5iZiI6MTc3MzI1ODMzNywicGF0aCI6Ii8zMTk5NjAvMzY5Njk0OTM3LThiZWYxYTI2LTRiZmEtNGEzZS04YWYxLTc3NzlhNjhlMTgyOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMxMVQxOTQ1MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MmI3YTA4N2NmNzRmOWJmMDhlOTM2MmU0NTFmYmZkYzNhMjVlNDRhOTFmYTVjMjMzODUyODhhZjc0MTYwZTgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-bCwpyCKP1xcCWVKRzHGfU51LKCRXsiN2O-RhwV2Tj8">
> 
> Quelle: https://backend.jugendundsport.ch/fileservice/sdweb-docs-prod-jugdsport-files/files/2024/03/06/8b65a586-b642-44f8-920f-4e665fee56b4.pdf

immer noch.

Ich schlage folgende Implementierung vor:

<img width="1841" height="1689" alt="Image" src="https://github.com/user-attachments/assets/ec16df0d-d001-4deb-9670-69eb20e90d2a" />

Morgen, Mittag und Abend unterscheide ich jeweils nach geplanter Mahlzeit, dies automatisch aus dem Picasso zu lesen ist etwas umständlich. Daher schlage ich vor die Zeiten anpassbar zu machen.
Dasselbe bei der Identifikation von LS und LA Aktivitäten, diese haben je Kurs, Sprache und HL Kreativität unterschiedliche Präfixe.

Wenn der Check aktiviert ist, zeigen die Tage verschiedene Farben an:

Rot: Ungenügend J+S Zeit
Gelb: Genügen J+S Zeit, aber noch nicht korrekt auf die Tageszeiten verteilt
Grün: Bene

Die Einstellungen werden nicht persistiert, alles nur adhoc im Frontend.

**Comment by Im-a-Train** [comment]
Hi @manuelmeister 

Thank's for discussing the pr #9212 and your reply.

I think we don't need a such validation engine.
Keep it as simple as possible.

My approach was maybe a bit too much, for solving my actual problem.
As long as no correct skilled coach llm is implemented, which can do all the reading, validating and feedbacking stuff better than a human. I need to read, validate and feedback all the activities anyway. This is fine. 

But what is really annoying, is to sum up the correct J+S time in a three-week camp for every day...

So, small hidable statistic header could do the thing.
Sum up the total activity time by activity for every day. 

I could deliver a first prototype, if this would be an acceptable solution.

**Comment by chrisgwerder** [comment]
With https://github.com/ecamp/ecamp3/issues/10734 I think this issue is now obsolete.

**Comment by carlobeltrame** [comment]
I don't think so yet, unfortunately. There will still be rules for how to count daily sports and J+S activities, and the rules haven't really gotten simpler: Each J+S activity needs to report how much of it is sports, and you'll be tallying up these numbers as well as the total duration of (non-concurrent) J+S activities per day.

PBS has requested such an automatic counter, but this is not set nor specified fully yet. E.g. a request is to tally up per "column" in the picasso, but eCamp v3 does not have a concept of columns, we have a more advanced and flexible data model.

PBS has also requested some PBS-specific variations, so we are still gathering the requirements of various stakeholders before we decide how to implement this.

#9753 could probably enable people to implement their own simplified rules, but that is still work in progress too.
